<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 4: Bonus Exercise Results | Analysis Using R</title>
  <meta name="description" content="Principles of exploratory data analysis, RNAseq differential expression, and generalized linear models" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 4: Bonus Exercise Results | Analysis Using R" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Principles of exploratory data analysis, RNAseq differential expression, and generalized linear models" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 4: Bonus Exercise Results | Analysis Using R" />
  
  <meta name="twitter:description" content="Principles of exploratory data analysis, RNAseq differential expression, and generalized linear models" />
  

<meta name="author" content="Instructors: Shraddha Pai, Delaram Pouyabahar" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-4-finding-differentially-expressed-genes-from-rnaseq-data.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://bioinformaticsdotca.github.io/">
 <img src="images/bioinformatics_logo.png" width="90%" alt="bioinformatics.ca logo" style="display:block; margin-left:auto; margin-right:auto; margin-top:15px;">
 </a>
<li><center><strong><a href="./"><br>Analysis Using R<br><br></a></strong></center></li>
<a href="./">
 <img src="images/CBW_AUR_Icon.png" width="70%" alt="Analysis Using R logo" style="display:block; margin-left:auto; margin-right:auto; ">
 </a>
<br>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i>Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#meet-your-faculty"><i class="fa fa-check"></i>Meet your Faculty</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#pre-workshop-materials-and-laptop-setup-instructions"><i class="fa fa-check"></i>Pre-workshop Materials and Laptop Setup Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#laptop-setup-instructions"><i class="fa fa-check"></i>Laptop Setup Instructions</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-packages"><i class="fa fa-check"></i>R packages</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-example-data"><i class="fa fa-check"></i>Download example data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="lectures.html"><a href="lectures.html"><i class="fa fa-check"></i>Lectures</a></li>
<li class="chapter" data-level="" data-path="module-1-exploratory-data-analysis-and-clustering.html"><a href="module-1-exploratory-data-analysis-and-clustering.html"><i class="fa fa-check"></i>Module 1: Exploratory Data Analysis and Clustering</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-exploratory-data-analysis-and-clustering.html"><a href="module-1-exploratory-data-analysis-and-clustering.html#load-mouse-data"><i class="fa fa-check"></i>Load <code>mouse</code> data</a></li>
<li class="chapter" data-level="" data-path="module-1-exploratory-data-analysis-and-clustering.html"><a href="module-1-exploratory-data-analysis-and-clustering.html#correlations-distances-and-clustering"><i class="fa fa-check"></i>Correlations, distances, and clustering</a></li>
<li class="chapter" data-level="" data-path="module-1-exploratory-data-analysis-and-clustering.html"><a href="module-1-exploratory-data-analysis-and-clustering.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical clustering</a></li>
<li class="chapter" data-level="" data-path="module-1-exploratory-data-analysis-and-clustering.html"><a href="module-1-exploratory-data-analysis-and-clustering.html#k-means-clustering"><i class="fa fa-check"></i>K-means clustering</a></li>
<li class="chapter" data-level="" data-path="module-1-exploratory-data-analysis-and-clustering.html"><a href="module-1-exploratory-data-analysis-and-clustering.html#using-clvalid-to-determine-number-of-clusters"><i class="fa fa-check"></i>Using <code>clValid</code> to determine number of clusters</a></li>
<li class="chapter" data-level="" data-path="module-1-exploratory-data-analysis-and-clustering.html"><a href="module-1-exploratory-data-analysis-and-clustering.html#bonus-exercise"><i class="fa fa-check"></i>Bonus Exercise</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-1-bonus-exercise-results.html"><a href="module-1-bonus-exercise-results.html"><i class="fa fa-check"></i>Module 1: Bonus Exercise Results</a></li>
<li class="chapter" data-level="" data-path="module-2-dimensionality-reduction.html"><a href="module-2-dimensionality-reduction.html"><i class="fa fa-check"></i>Module 2: Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="" data-path="module-2-dimensionality-reduction.html"><a href="module-2-dimensionality-reduction.html#principal-component-analysis"><i class="fa fa-check"></i>Principal Component Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-2-dimensionality-reduction.html"><a href="module-2-dimensionality-reduction.html#step-1.-preparing-our-data"><i class="fa fa-check"></i>Step 1. Preparing Our Data</a></li>
<li class="chapter" data-level="" data-path="module-2-dimensionality-reduction.html"><a href="module-2-dimensionality-reduction.html#step-2.-apply-pca"><i class="fa fa-check"></i>Step 2. Apply PCA</a></li>
<li class="chapter" data-level="" data-path="module-2-dimensionality-reduction.html"><a href="module-2-dimensionality-reduction.html#step-3.-visualisation-of-pca-results"><i class="fa fa-check"></i>Step 3. Visualisation of PCA results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2-dimensionality-reduction.html"><a href="module-2-dimensionality-reduction.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i>t-Distributed Stochastic Neighbor Embedding (t-SNE)</a></li>
<li class="chapter" data-level="" data-path="module-2-dimensionality-reduction.html"><a href="module-2-dimensionality-reduction.html#uniform-manifold-approximation-and-projection-umap"><i class="fa fa-check"></i>Uniform Manifold Approximation and Projection (UMAP)</a></li>
<li class="chapter" data-level="" data-path="module-2-dimensionality-reduction.html"><a href="module-2-dimensionality-reduction.html#bonus-exercise-1"><i class="fa fa-check"></i>Bonus Exercise</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2-bonus-exercise-results.html"><a href="module-2-bonus-exercise-results.html"><i class="fa fa-check"></i>Module 2: Bonus Exercise Results</a></li>
<li class="chapter" data-level="" data-path="module-3-generalized-linear-models.html"><a href="module-3-generalized-linear-models.html"><i class="fa fa-check"></i>Module 3: Generalized Linear Models</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-generalized-linear-models.html"><a href="module-3-generalized-linear-models.html#essential-r-reading-tables-from-files-merging-basic-data-exploration"><i class="fa fa-check"></i>Essential R: Reading tables from files, merging, basic data exploration</a></li>
<li class="chapter" data-level="" data-path="module-3-generalized-linear-models.html"><a href="module-3-generalized-linear-models.html#explore-missing-data"><i class="fa fa-check"></i>Explore missing data</a></li>
<li class="chapter" data-level="" data-path="module-3-generalized-linear-models.html"><a href="module-3-generalized-linear-models.html#essential-r-plots-with-ggplot2"><i class="fa fa-check"></i>Essential R: Plots with <code>ggplot2</code></a></li>
<li class="chapter" data-level="" data-path="module-3-generalized-linear-models.html"><a href="module-3-generalized-linear-models.html#fit-binary-response-variable-using-glm-and-logistic-regression"><i class="fa fa-check"></i>Fit binary response variable using <code>glm()</code> and logistic regression</a></li>
<li class="chapter" data-level="" data-path="module-3-generalized-linear-models.html"><a href="module-3-generalized-linear-models.html#bonus-exercise-2"><i class="fa fa-check"></i>Bonus Exercise</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-bonus-exercise-results.html"><a href="module-3-bonus-exercise-results.html"><i class="fa fa-check"></i>Module 3: Bonus Exercise Results</a></li>
<li class="chapter" data-level="" data-path="module-4-finding-differentially-expressed-genes-from-rnaseq-data.html"><a href="module-4-finding-differentially-expressed-genes-from-rnaseq-data.html"><i class="fa fa-check"></i>Module 4: Finding differentially expressed genes from RNAseq data</a>
<ul>
<li class="chapter" data-level="" data-path="module-4-finding-differentially-expressed-genes-from-rnaseq-data.html"><a href="module-4-finding-differentially-expressed-genes-from-rnaseq-data.html#volcano-plot-r-base-graphics"><i class="fa fa-check"></i>Volcano plot (R base graphics)</a></li>
<li class="chapter" data-level="" data-path="module-4-finding-differentially-expressed-genes-from-rnaseq-data.html"><a href="module-4-finding-differentially-expressed-genes-from-rnaseq-data.html#volcano-plot-ggplot2"><i class="fa fa-check"></i>Volcano plot (ggplot2)</a></li>
<li class="chapter" data-level="" data-path="module-4-finding-differentially-expressed-genes-from-rnaseq-data.html"><a href="module-4-finding-differentially-expressed-genes-from-rnaseq-data.html#bonus-exercise-3"><i class="fa fa-check"></i>Bonus Exercise</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-bonus-exercise-results.html"><a href="module-4-bonus-exercise-results.html"><i class="fa fa-check"></i>Module 4: Bonus Exercise Results</a></li>
<li class="divider"></li>
<h1 id="sponsors">Sponsors</h1>
<center>
<a href="https://www.cihr-irsc.gc.ca"><img src="images/sponsors/cihr.png" width=90% style="padding-bottom: 20px" alt="Canadian Institutes of Health Research logo."></a>

<a href="https://genomecanada.ca/"><img src="images/sponsors/genomecanada.png" height="70" style="display:inline-block;"  style="padding-bottom: 20px" alt="Genome Canada logo.">
<a href="https://www.ontariogenomics.ca/"><img src="images/sponsors/ontariogenomics.png" height="70" style="display:inline-block;" style="padding-bottom: 20px;" alt="Ontario Genomics logo.">
<br><br>
<a href="https://oicr.on.ca/"><img src="images/sponsors/oicr.png" width=25% style="display:inline-block;" style="padding-bottom: 20px" alt="Ontario Institute for Cancer Research logo.">
<a href="https://www.hpcforhealth.ca/"><img src="images/sponsors/hpc4health.png" width=60% style="display:inline-block;" style="padding-bottom: 20px "alt="HPC4Health logo.">


<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysis Using R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="module-4-bonus-exercise-results" class="section level1 unnumbered hasAnchor">
<h1>Module 4: Bonus Exercise Results<a href="module-4-bonus-exercise-results.html#module-4-bonus-exercise-results" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb332"><pre class="sourceCode r codeblock"><code class="sourceCode r"><span id="cb332-1"><a href="module-4-bonus-exercise-results.html#cb332-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;yeastRNASeq&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) </span>
<span id="cb332-2"><a href="module-4-bonus-exercise-results.html#cb332-2" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;yeastRNASeq&quot;</span>)</span>
<span id="cb332-3"><a href="module-4-bonus-exercise-results.html#cb332-3" tabindex="-1"></a></span>
<span id="cb332-4"><a href="module-4-bonus-exercise-results.html#cb332-4" tabindex="-1"></a><span class="fu">library</span>(yeastRNASeq)</span>
<span id="cb332-5"><a href="module-4-bonus-exercise-results.html#cb332-5" tabindex="-1"></a></span>
<span id="cb332-6"><a href="module-4-bonus-exercise-results.html#cb332-6" tabindex="-1"></a><span class="fu">data</span>(geneLevelData)</span>
<span id="cb332-7"><a href="module-4-bonus-exercise-results.html#cb332-7" tabindex="-1"></a></span>
<span id="cb332-8"><a href="module-4-bonus-exercise-results.html#cb332-8" tabindex="-1"></a><span class="co"># only keep genes with cpm &gt; 25 in more than two samples</span></span>
<span id="cb332-9"><a href="module-4-bonus-exercise-results.html#cb332-9" tabindex="-1"></a></span>
<span id="cb332-10"><a href="module-4-bonus-exercise-results.html#cb332-10" tabindex="-1"></a><span class="co"># table of TRUE/FALSE values</span></span>
<span id="cb332-11"><a href="module-4-bonus-exercise-results.html#cb332-11" tabindex="-1"></a>genes_passing_cutoff <span class="ot">&lt;-</span> <span class="fu">cpm</span>(geneLevelData) <span class="sc">&gt;</span> <span class="dv">25</span> </span>
<span id="cb332-12"><a href="module-4-bonus-exercise-results.html#cb332-12" tabindex="-1"></a></span>
<span id="cb332-13"><a href="module-4-bonus-exercise-results.html#cb332-13" tabindex="-1"></a><span class="co"># For each gene, add up how many samples it is TRUE in (rowSums())</span></span>
<span id="cb332-14"><a href="module-4-bonus-exercise-results.html#cb332-14" tabindex="-1"></a><span class="co"># Then only keep those strictly greater than 2.</span></span>
<span id="cb332-15"><a href="module-4-bonus-exercise-results.html#cb332-15" tabindex="-1"></a>genes_keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(genes_passing_cutoff) <span class="sc">&gt;</span> <span class="dv">2</span></span>
<span id="cb332-16"><a href="module-4-bonus-exercise-results.html#cb332-16" tabindex="-1"></a></span>
<span id="cb332-17"><a href="module-4-bonus-exercise-results.html#cb332-17" tabindex="-1"></a><span class="co"># Finally subset the data</span></span>
<span id="cb332-18"><a href="module-4-bonus-exercise-results.html#cb332-18" tabindex="-1"></a>filtered <span class="ot">&lt;-</span> geneLevelData[genes_keep,]</span>
<span id="cb332-19"><a href="module-4-bonus-exercise-results.html#cb332-19" tabindex="-1"></a></span>
<span id="cb332-20"><a href="module-4-bonus-exercise-results.html#cb332-20" tabindex="-1"></a><span class="co"># How many genes pass this threshold?</span></span>
<span id="cb332-21"><a href="module-4-bonus-exercise-results.html#cb332-21" tabindex="-1"></a><span class="fu">str</span>(filtered)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    4552 obs. of  4 variables:
##  $ mut_1: num  38 31 55 189 33 23 51 50 40 39 ...
##  $ mut_2: num  39 33 52 180 41 13 52 48 47 40 ...
##  $ wt_1 : num  35 40 47 151 32 73 54 30 14 55 ...
##  $ wt_2 : num  34 26 47 180 29 63 45 32 25 70 ...</code></pre>
<div class="sourceCode" id="cb334"><pre class="sourceCode r codeblock"><code class="sourceCode r"><span id="cb334-1"><a href="module-4-bonus-exercise-results.html#cb334-1" tabindex="-1"></a><span class="co">#  as.matrix(filtered): the count data in the right class</span></span>
<span id="cb334-2"><a href="module-4-bonus-exercise-results.html#cb334-2" tabindex="-1"></a><span class="co"># phenoData: The sample information</span></span>
<span id="cb334-3"><a href="module-4-bonus-exercise-results.html#cb334-3" tabindex="-1"></a></span>
<span id="cb334-4"><a href="module-4-bonus-exercise-results.html#cb334-4" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;Mut&quot;</span>, <span class="st">&quot;WT&quot;</span>),<span class="at">each=</span><span class="dv">2</span>), </span>
<span id="cb334-5"><a href="module-4-bonus-exercise-results.html#cb334-5" tabindex="-1"></a>    <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;WT&quot;</span>,<span class="st">&quot;Mut&quot;</span>)) <span class="co"># we tell R to use &quot;WT&quot; as the reference</span></span>
<span id="cb334-6"><a href="module-4-bonus-exercise-results.html#cb334-6" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="fu">as.matrix</span>(filtered), </span>
<span id="cb334-7"><a href="module-4-bonus-exercise-results.html#cb334-7" tabindex="-1"></a>           <span class="at">group =</span> group)  </span>
<span id="cb334-8"><a href="module-4-bonus-exercise-results.html#cb334-8" tabindex="-1"></a></span>
<span id="cb334-9"><a href="module-4-bonus-exercise-results.html#cb334-9" tabindex="-1"></a><span class="do">## matrix of experimental design </span></span>
<span id="cb334-10"><a href="module-4-bonus-exercise-results.html#cb334-10" tabindex="-1"></a>mod <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>group, y)</span>
<span id="cb334-11"><a href="module-4-bonus-exercise-results.html#cb334-11" tabindex="-1"></a></span>
<span id="cb334-12"><a href="module-4-bonus-exercise-results.html#cb334-12" tabindex="-1"></a></span>
<span id="cb334-13"><a href="module-4-bonus-exercise-results.html#cb334-13" tabindex="-1"></a><span class="do">## Normalize data</span></span>
<span id="cb334-14"><a href="module-4-bonus-exercise-results.html#cb334-14" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(y, <span class="at">method =</span> <span class="st">&quot;upperquartile&quot;</span>)</span>
<span id="cb334-15"><a href="module-4-bonus-exercise-results.html#cb334-15" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(y, mod)</span>
<span id="cb334-16"><a href="module-4-bonus-exercise-results.html#cb334-16" tabindex="-1"></a></span>
<span id="cb334-17"><a href="module-4-bonus-exercise-results.html#cb334-17" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">glmFit</span>(y, mod)</span>
<span id="cb334-18"><a href="module-4-bonus-exercise-results.html#cb334-18" tabindex="-1"></a>lrt <span class="ot">=</span> <span class="fu">glmLRT</span>(fit, <span class="at">coef =</span> <span class="dv">2</span>)</span>
<span id="cb334-19"><a href="module-4-bonus-exercise-results.html#cb334-19" tabindex="-1"></a>diffEx2 <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(lrt, </span>
<span id="cb334-20"><a href="module-4-bonus-exercise-results.html#cb334-20" tabindex="-1"></a>    <span class="at">adjust.method=</span><span class="st">&quot;BH&quot;</span>, </span>
<span id="cb334-21"><a href="module-4-bonus-exercise-results.html#cb334-21" tabindex="-1"></a>    <span class="at">p.value=</span><span class="fl">0.05</span></span>
<span id="cb334-22"><a href="module-4-bonus-exercise-results.html#cb334-22" tabindex="-1"></a>)</span>
<span id="cb334-23"><a href="module-4-bonus-exercise-results.html#cb334-23" tabindex="-1"></a><span class="fu">table</span>(diffEx2)</span></code></pre></div>
<pre><code>## diffEx2
##   -1    0    1 
## 1049 2209 1294</code></pre>
<div class="sourceCode" id="cb336"><pre class="sourceCode r codeblock"><code class="sourceCode r"><span id="cb336-1"><a href="module-4-bonus-exercise-results.html#cb336-1" tabindex="-1"></a>DEGS <span class="ot">=</span> <span class="fu">topTags</span>(lrt, <span class="at">n=</span><span class="fu">nrow</span>(y))<span class="sc">$</span>table</span></code></pre></div>
<div class="sourceCode" id="cb337"><pre class="sourceCode r codeblock"><code class="sourceCode r"><span id="cb337-1"><a href="module-4-bonus-exercise-results.html#cb337-1" tabindex="-1"></a><span class="do">## check out differentially expressed genes</span></span>
<span id="cb337-2"><a href="module-4-bonus-exercise-results.html#cb337-2" tabindex="-1"></a><span class="fu">head</span>(DEGS)</span></code></pre></div>
<pre><code>##            logFC   logCPM       LR PValue FDR
## snR10   6.268938 11.39811 4545.409      0   0
## YGL088W 6.195050 11.32542 4314.679      0   0
## snR18   6.183593 10.98786 3453.878      0   0
## snR35   6.078012 10.50997 2221.059      0   0
## snR9    5.769437 11.19751 3846.696      0   0
## snR8    5.562905 10.34273 2140.791      0   0</code></pre>
<div class="sourceCode" id="cb339"><pre class="sourceCode r codeblock"><code class="sourceCode r"><span id="cb339-1"><a href="module-4-bonus-exercise-results.html#cb339-1" tabindex="-1"></a>DEGS_sig <span class="ot">=</span> DEGS[DEGS<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span>
<span id="cb339-2"><a href="module-4-bonus-exercise-results.html#cb339-2" tabindex="-1"></a><span class="fu">head</span>(DEGS_sig)</span></code></pre></div>
<pre><code>##            logFC   logCPM       LR PValue FDR
## snR10   6.268938 11.39811 4545.409      0   0
## YGL088W 6.195050 11.32542 4314.679      0   0
## snR18   6.183593 10.98786 3453.878      0   0
## snR35   6.078012 10.50997 2221.059      0   0
## snR9    5.769437 11.19751 3846.696      0   0
## snR8    5.562905 10.34273 2140.791      0   0</code></pre>
<div class="sourceCode" id="cb341"><pre class="sourceCode r codeblock"><code class="sourceCode r"><span id="cb341-1"><a href="module-4-bonus-exercise-results.html#cb341-1" tabindex="-1"></a><span class="fu">dim</span>(DEGS_sig)</span></code></pre></div>
<pre><code>## [1] 2343    5</code></pre>
<p>Now merge the <code>DEG</code> and table with the table with the table that has the up/downregulated status:</p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r codeblock"><code class="sourceCode r"><span id="cb343-1"><a href="module-4-bonus-exercise-results.html#cb343-1" tabindex="-1"></a><span class="co"># needed to use merge function</span></span>
<span id="cb343-2"><a href="module-4-bonus-exercise-results.html#cb343-2" tabindex="-1"></a>diffEx2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(diffEx2) </span>
<span id="cb343-3"><a href="module-4-bonus-exercise-results.html#cb343-3" tabindex="-1"></a><span class="co"># give column more intuitive name</span></span>
<span id="cb343-4"><a href="module-4-bonus-exercise-results.html#cb343-4" tabindex="-1"></a><span class="fu">colnames</span>(diffEx2)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;gene_status&quot;</span></span>
<span id="cb343-5"><a href="module-4-bonus-exercise-results.html#cb343-5" tabindex="-1"></a></span>
<span id="cb343-6"><a href="module-4-bonus-exercise-results.html#cb343-6" tabindex="-1"></a><span class="co"># add the common &quot;gene&quot; column to merge the two tables</span></span>
<span id="cb343-7"><a href="module-4-bonus-exercise-results.html#cb343-7" tabindex="-1"></a>diffEx2<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(diffEx2)</span>
<span id="cb343-8"><a href="module-4-bonus-exercise-results.html#cb343-8" tabindex="-1"></a>DEGS<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(DEGS)</span>
<span id="cb343-9"><a href="module-4-bonus-exercise-results.html#cb343-9" tabindex="-1"></a></span>
<span id="cb343-10"><a href="module-4-bonus-exercise-results.html#cb343-10" tabindex="-1"></a>mega <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="at">x =</span> DEGS, <span class="at">y =</span> diffEx2, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span></code></pre></div>
<p>Assign the column name for gene status:</p>
<div class="sourceCode" id="cb344"><pre class="sourceCode r codeblock"><code class="sourceCode r"><span id="cb344-1"><a href="module-4-bonus-exercise-results.html#cb344-1" tabindex="-1"></a>is_sig <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;n.s.&quot;</span>, <span class="fu">nrow</span>(mega)) <span class="co"># default is ns</span></span>
<span id="cb344-2"><a href="module-4-bonus-exercise-results.html#cb344-2" tabindex="-1"></a>is_sig[<span class="fu">which</span>(mega<span class="sc">$</span>gene_status <span class="sc">&gt;</span> <span class="dv">0</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;Upregulated&quot;</span></span>
<span id="cb344-3"><a href="module-4-bonus-exercise-results.html#cb344-3" tabindex="-1"></a>is_sig[<span class="fu">which</span>(mega<span class="sc">$</span>gene_status <span class="sc">&lt;</span> <span class="dv">0</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;Downregulated&quot;</span></span>
<span id="cb344-4"><a href="module-4-bonus-exercise-results.html#cb344-4" tabindex="-1"></a></span>
<span id="cb344-5"><a href="module-4-bonus-exercise-results.html#cb344-5" tabindex="-1"></a><span class="co"># use levels() to tell R how to order the categorical </span></span>
<span id="cb344-6"><a href="module-4-bonus-exercise-results.html#cb344-6" tabindex="-1"></a><span class="co"># variables. Downregulated = 1, n.s.=2, and Upregulated=3. </span></span>
<span id="cb344-7"><a href="module-4-bonus-exercise-results.html#cb344-7" tabindex="-1"></a><span class="co"># By default, R orders categorical variables alphabetically,</span></span>
<span id="cb344-8"><a href="module-4-bonus-exercise-results.html#cb344-8" tabindex="-1"></a><span class="co"># which may not make sense!</span></span>
<span id="cb344-9"><a href="module-4-bonus-exercise-results.html#cb344-9" tabindex="-1"></a>mega<span class="sc">$</span>is_sig <span class="ot">&lt;-</span> <span class="fu">factor</span>(is_sig, </span>
<span id="cb344-10"><a href="module-4-bonus-exercise-results.html#cb344-10" tabindex="-1"></a>    <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Downregulated&quot;</span>,<span class="st">&quot;n.s.&quot;</span>,<span class="st">&quot;Upregulated&quot;</span>))</span></code></pre></div>
<p>Finally create the volcano plot:</p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r codeblock"><code class="sourceCode r"><span id="cb345-1"><a href="module-4-bonus-exercise-results.html#cb345-1" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mega, </span>
<span id="cb345-2"><a href="module-4-bonus-exercise-results.html#cb345-2" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span> <span class="co"># -log10 conversion  </span></span>
<span id="cb345-3"><a href="module-4-bonus-exercise-results.html#cb345-3" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color=</span>is_sig),<span class="at">size =</span> <span class="dv">2</span><span class="sc">/</span><span class="dv">5</span>) <span class="sc">+</span>  </span>
<span id="cb345-4"><a href="module-4-bonus-exercise-results.html#cb345-4" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="fu">expression</span>(<span class="st">&quot;log&quot;</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">&quot;FC&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb345-5"><a href="module-4-bonus-exercise-results.html#cb345-5" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="st">&quot;-log&quot;</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">&quot;FDR&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb345-6"><a href="module-4-bonus-exercise-results.html#cb345-6" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(</span>
<span id="cb345-7"><a href="module-4-bonus-exercise-results.html#cb345-7" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;dodgerblue3&quot;</span>, <span class="st">&quot;gray50&quot;</span>, <span class="st">&quot;firebrick3&quot;</span>)) </span>
<span id="cb345-8"><a href="module-4-bonus-exercise-results.html#cb345-8" tabindex="-1"></a></span>
<span id="cb345-9"><a href="module-4-bonus-exercise-results.html#cb345-9" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-90-1.png" width="672" /></p>


</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-4-finding-differentially-expressed-genes-from-rnaseq-data.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": null,
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
